src/components/button/types.ts

```diff
@@ -0,0 +1,16 @@
+import { FormEvent, HTMLAttributes } from 'react';
+
+enum ButtonType {
+  Type1 = 1,
+  Type2 = 2,
+  Type3 = 3,
+  Type4 = 4,
+  Type5 = 5,
+  Type6 = 6,
+}
+export interface IProps extends HTMLAttributes<HTMLButtonElement> {
+  label: string;
+  type?: ButtonType;
+  onClick?: (() => void) | ((e: FormEvent) => void);
+  className?: string;
```

Comment:
버튼 요소를 상속 받아서 onClick은 명시하지 않아도 될 것 같아요!
혹시 label 대신에 childeren으로 처리하신다면 label도 없어도 되지 않을까 싶습니당ㅎㅎ

```suggestion
export interface IProps extends HTMLAttributes<HTMLButtonElement> {
  disabled?: boolean;
  type?: 1 | 2 | 3 | 4 | 5 | 6;
}
```